cmd/ocsp-responder/main.go

```diff
@@ -79,19 +80,35 @@ func (src *DBSource) Response(req *ocsp.Request) ([]byte, bool) {
 	serialString := core.SerialToString(req.SerialNumber)
 	log.Debug(fmt.Sprintf("Searching for OCSP issued by us for serial %s", serialString))
 
-	var ocspResponse core.OCSPResponse
+	var response []byte
+	defer func() {
+		if len(response) != 0 {
+			log.Info(fmt.Sprintf("OCSP Response sent for CA=%s, Serial=%s", hex.EncodeToString(src.caKeyHash), serialString))
+		}
+	}()
 	// Note: we order by id rather than createdAt, because otherwise we sometimes
 	// get the wrong result if a certificate is revoked in the same second as its
 	// last update (e.g. client issues and instant revokes).
-	err := src.dbMap.SelectOne(&ocspResponse, "SELECT * from ocspResponses WHERE serial = :serial ORDER BY id DESC LIMIT 1;",
-		map[string]interface{}{"serial": serialString})
-	if err != nil {
+	err := src.dbMap.SelectOne(
+		&response,
+		"SELECT ocspResponse FROM certificateStatus WHERE serial = :serial",
+		map[string]interface{}{"serial": serialString},
+	)
+	if err != nil || len(response) == 0 {
+		if err == sql.ErrNoRows || len(response) == 0 {
```

Comment:
ErrNoRows is not a reason to retry in the ocspResponses table: if the certificate exists, its certificateStatus should exist (we have a transaction to ensure this). So I think you can do a single level of `if len(response) == 0`. Since you're assigning to an `err` in the parent scope, you can share a single `if err != nil` after that.

```
if